tests/Doctrine/Tests/DBAL/Functional/ConnectionTest.php

```diff
@@ -70,6 +70,40 @@ public function testTransactionNestingBehavior() : void
             $this->connection->rollBack();
             self::assertEquals(0, $this->connection->getTransactionNestingLevel());
         }
+
+        $this->connection->beginTransaction();
+        $this->connection->close();
+        $this->connection->beginTransaction();
+        self::assertEquals(1, $this->connection->getTransactionNestingLevel());
+    }
+
+    public function testTransactionNestingLevelIsResetOnReconnect() : void
+    {
+        if ($this->connection->getDatabasePlatform()->getName() === 'sqlite') {
+            $params           = $this->connection->getParams();
+            $params['memory'] = false;
+            $params['path']   = '/tmp/test_nesting.sqlite';
+
+            $connection = DriverManager::getConnection(
+                $params,
+                $this->connection->getConfiguration(),
+                $this->connection->getEventManager()
+            );
+        } else {
+            $connection = $this->connection;
+        }
+
+        $connection->executeQuery('CREATE TABLE test_nesting(test int not null)');
+
+        $this->connection->beginTransaction();
+        $this->connection->beginTransaction();
+        $connection->close(); // connection closed in runtime (for example if lost or another application logic)
+
+        $connection->beginTransaction(); // should connect, reset nesting level and increase it once
```

Comment:
The comment doesn't seem relevant to what the test is about.

Reply:
"Should realy start the transaction, and the rollback() call below realy leads to rollback it "

Reply:
The transaction nesting level could be checked here too.

Reply:
> "Should realy start the transaction, and the rollback() call below realy leads to rollback it " - better?

Just remove the comment. The previous one says enough.

Reply:
Ok,. done

Reply:
Hm… it's still there.

Reply:
Hm, it really was. Forgot to commit, sorry. Now it's realy done)